src/model/page-config-resolver.js

```diff
@@ -291,6 +358,64 @@ class MicrodataParser {
     this.win_ = win;
   }
 
+  discoverAccess_(prop) {
+    if (prop.getAttribute('itemprop') == 'isAccessibleForFree') {
+      const value = prop.content || prop.textContent;
+      if (typeof value == 'boolean') {
+        return value;
+      }
+		  if (value === 'False' || value === 'false') {
+			  return false;
+      } else if (value === 'true' || value == 'True') {
+        return true;
+      }
+    }
+    return null;
+  }
+
+  discoverProductId_(prop) {
+    if (prop.getAttribute('itemprop') == 'productID')	{
+      return prop.content || prop.textContent;
+    }
+    return null;
+  };
+
+  tryExtractConfig_(doc) {
+    let itemId = 0;
+    const results = [];
+    doc.querySelectorAll('[itemscope]')
+        .forEach(element => {
+          if (element.getAttribute('itemtype') == 'http://schema.org/NewsArticle') {
```

Comment:
Can we combine `itemtype` with selector above? That'd be a lot faster.

Reply:
What if itemscope is defined in the parent div and itemtype is defined as a meta of a child div? We will end up missing it.

Reply:
But that's not what you're doing now, is it?